services/task-svc/internal/task/task.go

```diff
@@ -578,6 +579,14 @@ func (ServiceServer) AssignTaskToUser(ctx context.Context, req *pb.AssignTaskToU
 		Str("userID", userId.String()).
 		Msg("user assigned")
 
+	if currentUserId, err := common.GetUserID(ctx); err != nil {
+		if currentUserId == userId {
+			_ = events.DispatchTaskSelfAssignedEvent(ctx, id, userId)
+		} else {
+			_ = events.DispatchTaskAssignedEvent(ctx, id, userId)
+		}
+	}
+
```

Comment:
You are dispatching the events in the error handler here

Reply:
Good catch!